CMakeLists.txt

```diff
@@ -154,7 +154,12 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
 endif()
 
 # we are making *a Python 3.6 Interpreter* a required dependency, so find it here
-find_package(PythonInterp 3.6 REQUIRED)
+# If LINK_WITH_PYTHON, also request the Development (libs) at the same time, to ensure consistent version between interpreter and Development
+if(LINK_WITH_PYTHON)
+  find_package(Python 3.6 COMPONENTS Interpreter Development REQUIRED)
+else()
+  find_package(Python 3.6 COMPONENTS Interpreter REQUIRED)
+endif()
```

Comment:
New way, find both components at the same time to ensure consistent versions

Reply:
Love it!